### PR TITLE
Removing lua type checks from the render library

### PR DESCRIPTION
**Pull request by Manonox**
*Performance, yada-yada*
Idea was born here: https://github.com/thegrb93/StarfallEx/pull/1414#issuecomment-1383408358

### CODE REVIEW

**Comment by thegrb93** [comment]
Can't remove them all because lua-erroring midway in some of these functions will break the render lib.

**Comment by Manonox** [comment]
Most look fine to be removed, but there's some that can't be removed. pushMatrix and pushViewMatrix for example.
Can you list them all so I can revert those?

> Would also be better to remove the check rather than comment them.
Yeah, it's temporary, for stuff like ^

File: lua/starfall/libs_cl/render.lua [hunk 1]
```diff
@@ -2346,10 +2346,10 @@ end
 -- @param number w The width of the screen
 -- @param number h The height of the screen
 function render_library.setScreenDimensions(screen, x, y, w, h)
-	checkluatype(x, TYPE_NUMBER)
-	checkluatype(y, TYPE_NUMBER)
-	checkluatype(w, TYPE_NUMBER)
-	checkluatype(h, TYPE_NUMBER)
+	-- checkluatype(x, TYPE_NUMBER)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:2349]: These should stay

File: lua/starfall/libs_cl/render.lua [hunk 2]
```diff
@@ -2332,7 +2332,7 @@ render_library.getHDREnabled = render.GetHDREnabled
 function render_library.setChipOverlay(name)
 	local rt
 	if name~=nil then
-		checkluatype(name, TYPE_STRING)
+		-- checkluatype(name, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:2335]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 3]
```diff
@@ -2041,44 +2041,44 @@ end
 --- Renders the scene with the specified viewData to the current active render target.
 -- @param table tbl view The view data to be used in the rendering. See http://wiki.facepunch.com/gmod/Structures/ViewData. There's an additional key drawviewer used to tell the engine whether the local player model should be rendered.
 function render_library.renderView(tbl)
-	checkluatype(tbl, TYPE_TABLE)
+	-- checkluatype(tbl, TYPE_TABLE)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:2044]: These should stay

File: lua/starfall/libs_cl/render.lua [hunk 4]
```diff
@@ -1357,14 +1357,14 @@ end
...
 	if not renderdata.isRendering then SF.Throw("Not in rendering hook.", 2) end
-	checkluatype (x, TYPE_NUMBER)
-	checkluatype (y, TYPE_NUMBER)
-	checkluatype (w, TYPE_NUMBER)
-	checkluatype (h, TYPE_NUMBER)
-	checkluatype (startU, TYPE_NUMBER)
-	checkluatype (startV, TYPE_NUMBER)
-	checkluatype (endU, TYPE_NUMBER)
-	checkluatype (endV, TYPE_NUMBER)
+	-- checkluatype (x, TYPE_NUMBER)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1360]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 5]
```diff
@@ -1159,7 +1159,7 @@ end
 --- Sets the texture filtering function when viewing a far texture
 -- @param number val The filter function to use http://wiki.facepunch.com/gmod/Enums/TEXFILTER
 function render_library.setFilterMin(val)
-	checkluatype (val, TYPE_NUMBER)
+	-- checkluatype (val, TYPE_NUMBER)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1162]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 6]
```diff
@@ -1148,7 +1148,7 @@ end
 --- Sets the texture filtering function when viewing a close texture
 -- @param number val The filter function to use http://wiki.facepunch.com/gmod/Enums/TEXFILTER
 function render_library.setFilterMag(val)
-	checkluatype (val, TYPE_NUMBER)
+	-- checkluatype (val, TYPE_NUMBER)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1151]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 7]
```diff
@@ -1114,7 +1114,7 @@ function render_library.setRenderTargetTexture(name)
 		render.SetColorMaterial()
 		draw.NoTexture()
 	else
-		checkluatype (name, TYPE_STRING)
+		-- checkluatype (name, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1117]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 8]
```diff
@@ -1063,7 +1063,7 @@ end
 function render_library.selectRenderTarget(name)
 	if not renderdata.isRendering then SF.Throw("Not in rendering hook.", 2) end
 	if name then
-		checkluatype (name, TYPE_STRING)
+		-- checkluatype (name, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1066]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 9]
```diff
@@ -1024,15 +1024,15 @@ end
...
+	-- checkluatype (name, TYPE_STRING)
 	return renderdata.rendertargets[name] ~= nil
 end
 
 --- Creates a new render target to draw onto.
 -- The dimensions will always be 1024x1024
 -- @param string name The name of the render target
 function render_library.createRenderTarget(name)
-	checkluatype (name, TYPE_STRING)
+	-- checkluatype (name, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1035]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 10]
```diff
@@ -1024,15 +1024,15 @@ end
 --- Check if the specified render target exists.
 -- @param string name The name of the render target
 function render_library.renderTargetExists(name)
-	checkluatype (name, TYPE_STRING)
+	-- checkluatype (name, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:1027]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 11]
```diff
@@ -971,7 +971,7 @@ function render_library.setMaterialEffectColorModify(mat, cmStructure)
 	checkpermission(instance, nil, "render.effects")
 	if not renderdata.isRendering then SF.Throw("Not in rendering hook.", 2) end
 	local tex = gettexture(mat)
-	checkluatype(cmStructure, TYPE_TABLE)
+	-- checkluatype(cmStructure, TYPE_TABLE)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:974]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 12]
```diff
@@ -827,7 +827,7 @@ end
 -- @param function? done An optional callback called when the image is done loading. Passes the material, url
 -- @return Material The material. Use with render.setMaterial to draw with it.
 function render_library.createMaterial(tx, cb, done)
-	checkluatype (tx, TYPE_STRING)
+	-- checkluatype (tx, TYPE_STRING)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:830]: This should stay

File: lua/starfall/libs_cl/render.lua [hunk 13]
```diff
@@ -715,7 +715,7 @@ local viewmatrix_checktypes_ignore = {origin = true, angles = true}
 function render_library.pushViewMatrix(tbl)
 	if not renderdata.isRendering then SF.Throw("Not in rendering hook.", 2) end
 	if #view_matrix_stack == MATRIX_STACK_LIMIT then SF.Throw("Pushed too many matrices", 2) end
-	checkluatype(tbl, TYPE_TABLE)
+	-- checkluatype(tbl, TYPE_TABLE)
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua:718]: This should stay

**Comment by Manonox** [comment]
@thegrb93 Wait, fuckies, didn't see the first table check in renderView

**Comment by Manonox** [comment]
> @thegrb93 Wait, fuckies, didn't see the first table check in renderView

(The conversation block separated all checks except the last one)
I think it shouldn't be a problem, it will still error when it'll try to index the table

**Pull request merged**